management-interface.go

```diff
@@ -142,14 +150,51 @@ func IndexHandler(w http.ResponseWriter, r *http.Request) {
 
 // NetworkInterfacesHandler - Show the status of each newtwork interface
 func NetworkInterfacesHandler(w http.ResponseWriter, r *http.Request) {
-	data, err := AvailableInterfaces()
+	ifaces, err := net.Interfaces()
+	interfaces := []net.Interface{}
 	if err != nil {
-		log.Fatal(err)
+		log.Output(1, err.Error())
+	} else {
+		// Filter out loopback interfaces
+		for _, iface := range ifaces {
+			if iface.Flags&net.FlagLoopback == 0 {
+				// Not a loopback interface
+				interfaces = append(interfaces, iface)
+			}
+		}
 	}
-	// Need to put our output string in a struct so we can access it from html
-	outputStruct := MultiLineStringToBeDisplayed{Strings: data}
 
-	tmpl.ExecuteTemplate(w, "network-interfaces.html", outputStruct)
+	// Need to respond to individual requests to test if a network status is up or down.
+
+	tmpl.ExecuteTemplate(w, "network-interfaces.html", interfaces)
+}
+
+func CheckInterfaceHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	response := make(map[string]string)
+	// Extract interface id
+	interfaceVar := mux.Vars(r)["id"]
+	interfaceId, _ := strconv.Atoi(interfaceVar)
+	// Lookup interface by id
+	iface, err := net.InterfaceByIndex(interfaceId)
```

Comment:
Network interfaces can come and go so addressing them by integer id is perhaps not the best idea. What about using the interface name?

Reply:
Yes I think you're right.  Will change it.